glean-core/src/upload/request.rs

```diff
@@ -57,6 +80,10 @@ impl PingRequest {
             "Content-Type",
             "application/json; charset=utf-8".to_string(),
         );
+        if is_gzipped {
+            // TODO: Do we need "Content-Length"?
```

Comment:
We don't depend on it.
In pingsender we don't set it because we're effectively streaming the encoding, so we don't even know the size ahead of time.
As we do know it here, we could set it.